src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py

```diff
@@ -568,7 +571,7 @@ def test_role_assignment_handle_conflicted_assignments(self, resource_group):
 
                 self.assertGreaterEqual(len(local_defaults_config), 1)
                 actual = set([(x['name'], x['source'], x['value']) for x in local_defaults_config if x['name'] == 'group'])
-                expected = set([('group', os.path.join(temp_dir, '.azure', 'config'), self.kwargs['rg'])])
+                expected = set([('group', os.path.join(temp_dir, os.path.basename(self.cli_ctx.config.config_dir), 'config'), self.kwargs['rg'])])
```

Comment:
The config path is `temp_dir+basese(config_dir)+config` when `--scope local`

Reply:
Let's provide some concrete example.

Reply:
If work dir is `C:\Users\hanglei\AppData\Local\Temp\tmplxf_89ez`, local config file is `C:\Users\hanglei\AppData\Local\Temp\tmplxf_89ez\0azXbKR9OdJuZPFS\config`

`0azXbKR9OdJuZPFS` is the base name of config file: `~/.azure/dummy_cli_config_dir/0azXbKR9OdJuZPFS/config`

Reply:
The local config folder was constructed by `knack.config.CLIConfig`:

https://github.com/microsoft/knack/blob/e496c9590792572e680cb3ec959db175d9ba85dd/knack/config.py#L63

```py
config_dir_name = os.path.basename(self.config_dir)
current_config_dir = os.path.join(current_dir, config_dir_name)
```

`current_config_dir` is constructed by combining `current_dir` with the random dir name, making the path a little bit weird.